sdk/framework/openapi.go

```diff
@@ -534,31 +530,624 @@ func documentPath(p *Path, specialPaths *logical.Paths, requestResponsePrefix st
 	return nil
 }
 
-// constructRequestResponseName joins the given path with prefix & suffix into
-// a CamelCase request or response name.
+// When constructing a request or response name from a prefix + path, certain
+// paths result in very long names or names with duplicate parts. For such paths,
+// we use custom path mappings instead to generate request/response names.
+type knownPathKey struct {
```

Comment:
can we move this to a diff file?

Reply:
^agreed

Reply:
I kind of like it to be next to the function but we can move it out as well.
- What would be a good file name (something like `openapi_known_paths.go`)?
- Should we move constructRequestResponseIdentifier into the new file as well?

Reply:
I was just thinking moving the mappings into a different file if possible. I think your suggestion works or `openapi_path_mappings.go`